### PR TITLE
Bump up commons-compress to 1.26.1 to fix CVE 

### PR DESCRIPTION
**Pull request by sandeshkr419**
### Description
Follow-up on https://github.com/opensearch-project/OpenSearch/pull/12604, Bump common-compress package to 1.26.1 to fix CVE:

https://nvd.nist.gov/vuln/detail/CVE-2024-26308 NVD / Published Date:02/19/2024
https://nvd.nist.gov/vuln/detail/CVE-2024-25710 NVD / Published Date: 02/19/2024

Tika has been upgraded to latest release since any previous release were failing with required commons-compress upgrade. Also, other dependencies with ingest-attachement (with Tika) were upgraded to resolve minor future CVEs and test errors.

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by sandeshkr419** [comment]
Failing with `ingest-attachment` test failures right now. Upon upgrading `commons-compress` library, the `org.opensearch.ingest.attachment.TikaDocTests.testFiles` is failing since Tika is not able to parse files with extensions `.pages` & `.key`.

Tried upgrading tika libraries, did not succeed.

File: plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java [hunk 1]
```diff
@@ -65,6 +65,11 @@ public void testFiles() throws Exception {
 
         try (DirectoryStream<Path> stream = Files.newDirectoryStream(tmp)) {
             for (Path doc : stream) {
+                // Testing change with skipping - will remove in final code
```
Comment (reta) [inline plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:68]: Should be removed, right?

Comment (sandeshkr419) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java:68]: Yes. I skipped some test cases to isolate the failure points. I added `[Do not merge now]` to the title for the same reason. Will revert this.

**Comment by sandeshkr419** [comment]
I see 2 paths forward as of now to resolve the failures with `IWorkPackageParser.parse`:

1. Identify the issues with existing code flow without upgrading tika libraries - the dependencies failing are out of OpenSearch and I was not able to get to change implementations of how we use tika to resolve those failures.
2. Upgrading tika libraries to 2.9.1 - Not guaranteed to resolve the failures - however, facing issues with upgrading `tika-parsers-standard-package` as the [here](https://github.com/opensearch-project/OpenSearch/blob/main/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java#L92) are not getting resolved as expected as per related tika [issue](https://github.com/apache/tika/pull/1130). 

I am seeking clarifications from tika contributors to see if I'm missing anything.

**Comment by sandeshkr419** [comment]
**Quick update:** I am able to update the tika dependencies to 2.9.1 now, however the apple parsers (.key, .pages) are still failing due to a known issue with tika upon upgrading commons-compress 1.26.0. The fix will be available in 2.9.2, which is yet to be released.

Ref: https://github.com/apache/tika/pull/1130

**Review by kotwanikunal — APPROVED** [review]

**Comment by wbeckler** [comment]
The timing of the tika release might not align with the opensearch release process. Is there an option to build with a tika snapshot?

**Comment by sandeshkr419** [comment]
Fix in Tika merged: https://issues.apache.org/jira/browse/COMPRESS-661
To be released with Tika 2.9.2 next.

File: plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java (hunk 1 above)
Comment (sandeshkr419) [reply plugins/ingest-attachment/src/test/java/org/opensearch/ingest/attachment/TikaDocTests.java]: Tika updated now. Reverted this test/temporary change back to original. Resolving this thread.

**Comment by sandeshkr419** [comment]
Changes are ready for review!

**Review by kotwanikunal — APPROVED** [review]

File: plugins/ingest-attachment/build.gradle [hunk 2]
```diff
@@ -50,14 +50,24 @@ dependencies {
   api "org.apache.tika:tika-parsers:${versions.tika}"
   // Required for the various document parsers
   api "org.apache.tika:tika-parsers-standard-package:${versions.tika}"
+  api "org.apache.tika:tika-parser-apple-module:${versions.tika}"
+  api "org.apache.tika:tika-parser-html-module:${versions.tika}"
+  api "org.apache.tika:tika-parser-microsoft-module:${versions.tika}"
```
Comment (reta) [inline plugins/ingest-attachment/build.gradle:55]: @sandeshkr419 why do we need all these Tika parsers? We didn't bundle them before

Comment (sandeshkr419) [reply plugins/ingest-attachment/build.gradle:55]: Yes, the tika parsers were not being packaged from tika 2.8 onwards.
https://github.com/apache/tika/pull/1130

I had to manually add them to resolve the below dependency failures after upgrading to tika 2.8+. 

```
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:94: error: package org.apache.tika.parser.html does not exist
        new org.apache.tika.parser.html.HtmlParser(),
                                       ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:95: error: package org.apache.tika.parser.pdf does not exist
        new org.apache.tika.parser.pdf.PDFParser(),
                                      ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:96: error: package org.apache.tika.parser.txt does not exist
        new org.apache.tika.parser.txt.TXTParser(),
                                      ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:97: error: package org.apache.tika.parser.microsoft.rtf does not exist
        new org.apache.tika.parser.microsoft.rtf.RTFParser(),
                                                ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:98: error: package org.apache.tika.parser.microsoft does not exist
        new org.apache.tika.parser.microsoft.OfficeParser(),
                                            ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:99: error: package org.apache.tika.parser.microsoft does not exist
        new org.apache.tika.parser.microsoft.OldExcelParser(),
                                            ^
/Users/kusandes/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:100: error: package org.apache.tika.parser.microsoft.ooxml does not exist
        ParserDecorator.withoutTypes(new org.apache.tika.parser.microsoft.ooxml.OOXMLParser(), EXCLUDES),
                                                                               ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:101: error: package org.apache.tika.parser.odf does not exist
        new org.apache.tika.parser.odf.OpenDocumentParser(),
                                      ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:102: error: package org.apache.tika.parser.iwork does not exist
        new org.apache.tika.parser.iwork.IWorkPackageParser(),
                                        ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:103: error: package org.apache.tika.parser.xml does not exist
        new org.apache.tika.parser.xml.DcXMLParser(),
                                      ^
/Users/--/workplace/opensearch/OpenSearch/plugins/ingest-attachment/src/main/java/org/opensearch/ingest/attachment/TikaImpl.java:104: error: package org.apache.tika.parser.epub does not exist
        new org.apache.tika.parser.epub.EpubParser(), };
                                       ^
```

Comment (reta) [reply plugins/ingest-attachment/build.gradle:55]: Thank you

**Pull request merged**